src/java.base/unix/native/libnet/NetworkInterface.c

```diff
@@ -769,6 +770,7 @@ static jobject createNetworkInterface(JNIEnv *env, netif *ifs) {
                     } else {
                         return NULL;
                     }
+                    (*env)->DeleteLocalRef(env, ia2Obj);
```

Comment:
The pattern is 1) SetObjectXXX, followed by 2) DeleteLocalRef (once there is another, non-local, reference). It might be worth relocating this line of code to directly after the SetObjectField.